doc/user/content/sql/alter-sink.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "ALTER SINK"
+description: "`ALTER SINK` changes certain characteristics of a sink."
+menu:
+  main:
+    parent: 'commands'
+---
+
+`ALTER SINK` changes certain characteristics of a sink.
+
+## Syntax
+
+{{< diagram "alter-sink.svg" >}}
+
+## Context
+
+Altering the `FROM` item of a sink results in Materialize picking a consistent
+cutover timestamp. The resulting topic will contain all the updates of the old
+item that happened before the cutover timestamp and all the updates of the new
+item that happened after the cutover timestamp. Care must be taken to ensure
+that the old and the new items make sense when stitched together. See the
+anomalies section for more details on what can be observed.
+
+**Note**: In order to select a consistent timestamp Materialize waits for the
+previous definition of the sink to make enough progress such that there is
+overlap between the available timestamps of the desired new item. Attempting to
+`ALTER` an unhealthy sink that can't make progress will result to the command
+timing out.
+
+### Valid schema changes
+
+Materialize allows altering a sink to a new item provided that the new item
+leads to a valid sink definition with the same rules as the `CREATE SINK`
+statement.
+
+When using the Avro format with a schema registry care must be taken such that
+the generated Avro schema of the new item is compatible with the previously
+published schema. When that is not the case the `ALTER SINK` statement will
+succeed but the subsequent execution of the sink will result in errors and will
+be unable to make progress.
+
+### Possible anomalies
```

Comment:
These are all standard operating procedure for a stream processor. I wonder if we want to call that out. We're "only" providing the same level of correctness as a sp here as opposed to our standard higher level of correctness.